Hibernate envers for db versioning (#203)
jpa-buddy/documentation: database-versioning/database-versioning.md

* `@TimeZoneStorage` & `@TimeZoneColumn`

## Hibernate Envers Support

<div class="note">
This functionality becomes available only if the <a href="https://jpa-buddy.com/documentation/#dependencies" target="_blank">corresponding dependency</a> is added to the project.
</div>

<a href="https://hibernate.org/orm/envers/" target="_blank">Hibernate Envers</a> is a module that enables easy auditing of entities in a database. JPA Buddy can generate all the required tables for Hibernate Envers to function correctly. This includes audit tables for entities marked with the `@Audited` annotation and revision table for the entity annotated with `@RevisionEntity` annotation. Check out how it works in action:

<div class="youtube">
<iframe width="560" height="315" src="https://www.youtube.com/embed/k5OTgQBmSrg" title="YouTube video player" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
</div>

> JPA Buddy offers flexible settings for Hibernate Envers. You can find more information about these settings in the [relevant section](#hibernate-envers).

Reviewer: ```suggestion
JPA Buddy offers flexible settings for Hibernate Envers. For more details, refer to the [corresponding section](#hibernate-envers).
```

</table>
</div>

### Hibernate Envers

![hibernate-envers-settings](img/hibernate-envers-settings.png)

> Hibernate Envers offers a lot of customization options. You can specify the desired configuration in the JPA Buddy settings (2), or let JPA Buddy read the existing settings from `.properties` files automatically (1).

Reviewer: ```suggestion
Hibernate Envers offers various customization options. You can specify the desired configuration in the JPA Buddy settings (2), or let JPA Buddy read the existing settings from `.properties` files automatically (1).
```
